docs/doc/60-contributing/03-rfcs/20221013-iceberg-external-table.md

```diff
@@ -0,0 +1,291 @@
+---
+title: Iceberg External Table
+description: Accessing Apache Iceberg formatted data source as external table.
+---
+
+- RFC PR: [datafuselabs/databend#8215](https://github.com/datafuselabs/databend/pull/8215)
+- Tracking Issue: [datafuselabs/databend#8216](https://github.com/datafuselabs/databend/issues/8216)
+
+## Summary
+
+Iceberg is a widely supported table format among data lake-houses.
+This RFC describes how the iceberg external table will behave and how we will implement it.
+
+## Motivation
+
+The [Apache Iceberg](https://iceberg.apache.org) is a table format widely used in data lake-houses, it offers a more complete logical view of databases and has much higher performance of accessing as external tables, since users don't need to know about partitions and less indexing through files when accessing is required comparing to `Hive`.
+
+The splitted and well-defined structure of `Iceberg` also makes concurrent accesssing and version managementing on data sources safer, saver and convienient.
+
+![stack of Apache Iceberg](https://iceberg.apache.org/img/iceberg-metadata.png)
+
+Supporting Iceberg will empower databend as a open data lake, giving it more OLAP abilities.
+
+## Guide-level explanation
+
+### Create Iceberg Table
+
+To use Iceberg external table, users need to have an iceberg storage with neccessary permission established. Then they can establish a external table with `ICEBERG` engine:
+
+```sql
+CREATE EXTERNAL TABLE [IF NOT EXISTS] [db.]table_name
+[(
+    <column_name> <data_type> [ NOT NULL | NULL] [ { DEFAULT <expr> }],
+    <column_name> <data_type> [ NOT NULL | NULL] [ { DEFAULT <expr> }],
+    ...
+)] ENGINE=ICEBERG
+ENGINE_OPTIONS=(
```

Comment:
What's the DATABASE and TABLE here?

Reply:
Iceberg has built-in databases and tables in the catalog, but this RFC only proposes creating a single external table from Iceberg.

Reply:
If so, we can do it as a hive table in databend: new catalog and new table engine.

Reply:
> If so, we can do it as a hive table in databend: new catalog and new table engine.

Using a catalog in hive way is not good for me. We should allow users use to configure the catalog for the iceberg. For example, specify the iceberg location by SQL instead of static config.

In this way, we need to introduce a `create catalog` like:

```sql
CREATE CATALOG my_icerberg TYPE='iceberg' CONNECTION=(URL='s3://my_bucket')
```

Reply:
Good question. 

The `catalog` can be a table option or others? Preferred not to let the user configure them in the `databend-query.toml`, so the databend cloud user can use databend as a data lake with `hive+hdfs` / `iceberg` storage.

Reply:
👍 
the `CREATE CATALOG` is good to me, we can create a hive catalog too.

Reply:
Will reply on https://github.com/datafuselabs/databend/pull/8255